src/game/client/ingameui.cpp

```diff
@@ -1751,3 +1765,1230 @@ void InGameUI::Recreate_Control_Bar()
+                                display_name.Format(U_CHAR("%hs: %s"), team_name.Str(), display_name.Str());
+                            }
+                        } else if (team_name.Is_Empty()) {
+                            display_name.Format(U_CHAR("%hs: %s"), object_name.Str(), display_name.Str());
+                        } else {
+                            display_name.Format(
+                                U_CHAR("%hs(%hs): %s"), team_name.Str(), object_name.Str(), display_name.Str());
+                        }
+
+                        display_name.Format(U_CHAR("%s - %hs"), display_name.Str(), state_name.Str());
+                    }
+#endif
+
+                    Utf16String tooltip;
+                    static NameKeyType warehouseModuleKey = g_theNameKeyGenerator->Name_To_Key("SupplyWarehouseDockUpdate");
+                    SupplyWarehouseDockUpdate *warehouse =
+                        static_cast<SupplyWarehouseDockUpdate *>(object->Find_Update_Module(warehouseModuleKey));
+
+                    if (warehouse != nullptr) {
+                        tooltip.Format(g_theGameText->Fetch("TOOLTIP:SupplyWarehouse"),
+                            g_theWriteableGlobalData->m_valuesPerSupplyBox * warehouse->Get_Boxes_Stored());
+                        display_name.Concat(tooltip);
+                    }
+
+                    if (player != nullptr) {
+                        Utf16String player_display;
+
+                        if (g_theRecorder->Is_Multiplayer() && player->Is_Playable_Side()) {
+                            player_display.Format(
+                                U_CHAR("%s\n%s"), display_name.Str(), player->Get_Player_Display_Name().Str());
+                        } else {
+                            player_display = display_name;
+                        }
+
+                        int index;
+
+                        if (g_thePlayerList != nullptr) {
+                            index = g_thePlayerList->Get_Local_Player()->Get_Player_Index();
+                        } else {
+                            index = 0;
+                        }
+
+                        int x;
+                        int y;
+                        g_thePartitionManager->World_To_Cell(object->Get_Position()->x, object->Get_Position()->y, &x, &y);
+
+                        if (g_thePartitionManager->Get_Shroud_Status_For_Player(index, x, y) == SHROUD_STATUS_CLEAR) {
+                            RGBColor color;
+
+                            if (is_disguised) {
+                                color.Set_From_Int(player->Get_Color());
+                            } else {
+                                color.Set_From_Int(drawable->Get_Object()->Get_Indicator_Color());
+
+                                const Object *obj = drawable->Get_Object();
+
+                                if (obj != nullptr) {
+                                    ContainModuleInterface *contain2 = obj->Get_Contain();

```

Comment:
This is same as `contain`, no?

Reply:
Don't ask me why it does this but it does.